src/config/functions.ts

```diff
@@ -53,7 +53,23 @@ export const fnSchema = {
 			log.error(`Unable to get architecture: ${e}`);
 			return 'unknown';
 		}
-	},
+	}),
+	deviceType: _.memoize(async () => {
```

Comment:
`_.once` makes more sense if you never plan to pass arguments:
```suggestion
	deviceType: _.once(async () => {
```
There is the issue of if the promise fails whether you want to remember that failure and forever fail (in this case the try/catch should avoid that though). If you don't want to remember the failure then an easy solution is to use `memoizee` with `promise: true`, and the supervisor already uses memoizee elsewhere so it won't increase the size